pkg/aws_keyhub/role_selection.go

```diff
@@ -0,0 +1,59 @@
+package aws_keyhub
+
+import (
+	"errors"
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/sirupsen/logrus"
+	"strings"
+)
+
+func SelectRoleAndPrincipal(roleArn string, rolesAndPrincipals map[string]RolesAndPrincipals) RolesAndPrincipals {
+	if len(roleArn) > 0 {
+		rolesAndPrincipal, err := findRoleAndPrincipalByOption(roleArn, rolesAndPrincipals)
+		if err != nil {
+			return promptForRole(rolesAndPrincipals)
+		}
+		logrus.Infoln("Selected role", rolesAndPrincipal.Role, "based on -r parameter.")
+		return rolesAndPrincipal
+	}
+
+	return promptForRole(rolesAndPrincipals)
+}
+
+func promptForRole(rolesAndPrincipals map[string]RolesAndPrincipals) RolesAndPrincipals {
```

Comment:
Klopt het ook hier dat het lowercase begint?

Reply:
Functies in Go die met een hoofdletter beginnen zijn beschikbaar buiten het huidige package.